pysmt/smtlib/script.py

```diff
@@ -290,6 +333,43 @@ def __str__(self):
         return "\n".join((str(cmd) for cmd in self.commands))
 
 
+def _parse_goal(command, max_smt_goals=None, goal_position=None):
+    assert len(command.args) == 2, f"Command {command.name} must have 2 arguments"
+    singed = False
+    if len(command.args) >= 2:
+        options = command.args[1]
+        if options is not None:
+            for arg in options:
+                if arg[0] == ":signed":
+                    singed = arg[1]
+                break
+    if command.name == smtcmd.MAXIMIZE:
+        return MaximizationGoal(command.args[0], singed)
+    elif command.name == smtcmd.MINIMIZE:
+        return MinimizationGoal(command.args[0], singed)
+    elif command.name == smtcmd.MINMAX:
+        return MinMaxGoal(command.args[0], singed)
+    elif command.name == smtcmd.MAXMIN:
+        return MaxMinGoal(command.args[0], singed)
+    elif command.name == smtcmd.ASSERT_SOFT:
+        assert max_smt_goals is not None
+        assert goal_position is not None
+        formula = command.args[0]
+        cmd_annotations = dict(command.args[1])
+        max_smt_id = cmd_annotations.get(":id", "")
+        max_smt_weight = cmd_annotations.get(":weight", 1)
+        if not isinstance(max_smt_weight, int):
```

Comment:
??
According to the parser, `max_smt_weight` should be an `FNode`, never a raw number. We have to make sure this is globally true.